src/ORM/FieldType/DBField.php

```diff
@@ -136,6 +136,10 @@ abstract class DBField extends ViewableData implements DBIndexable
      */
     public function __construct($name = null, $options = [])
     {
+        if (preg_match("#[^\w]#", $name)) {
```

Comment:
I think this expression is a bit too restrictive. It's relatively common to define table field names like "field-with-dashes" or "field_with_underscores". While it's definitely not a common set up in the SilverStripe ORM, I don't think we should preclude it either.